app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardActivity.kt

```diff
@@ -1,30 +1,28 @@
 package id.rizmaulana.covid19.ui.overview
 
-import android.animation.ValueAnimator
-import android.graphics.Color
 import android.os.Bundle
-import android.widget.TextView
-import com.github.mikephil.charting.animation.Easing
-import com.github.mikephil.charting.data.PieData
-import com.github.mikephil.charting.data.PieDataSet
-import com.github.mikephil.charting.data.PieEntry
+import android.util.Log
+import android.view.View
 import id.rizmaulana.covid19.R
-import id.rizmaulana.covid19.data.model.CovidDaily
-import id.rizmaulana.covid19.data.model.CovidOverview
 import id.rizmaulana.covid19.databinding.ActivityDashboardBinding
-import id.rizmaulana.covid19.ui.adapter.DailyAdapter
+import id.rizmaulana.covid19.ui.adapter.VisitableRecyclerAdapter
+import id.rizmaulana.covid19.ui.adapter.viewholders.*
 import id.rizmaulana.covid19.ui.base.BaseActivity
+import id.rizmaulana.covid19.ui.base.BaseViewItem
 import id.rizmaulana.covid19.ui.detail.DetailActivity
 import id.rizmaulana.covid19.util.CaseType
-import id.rizmaulana.covid19.util.NumberUtils
-import id.rizmaulana.covid19.util.ext.color
 import id.rizmaulana.covid19.util.ext.observe
 import org.koin.android.viewmodel.ext.android.viewModel
 
 class DashboardActivity : BaseActivity() {
 
     private val viewModel by viewModel<DashboardViewModel>()
-    private val dailyAdapter by lazy { DailyAdapter() }
+    private val dailyAdapter by lazy { VisitableRecyclerAdapter(listOf(
+        DailyItemViewHolderFactory(),
+        OverviewItemViewHolderFactory(),
+        TextItemViewHolderFactory()
```

Comment:
it is could be more simplify with creating a single factory for those. imagine if you have a lot of `ViewHolderFactory`, you should declare all of them in this UI (or in another UI that used it too). for example, you can see my repo here: https://github.com/isfaaghyth/visitable-pattern-recyclerview

Reply:
Ohhh i see you're right, I've already looked at the example in your repo 👍 💯 
will try to implement it @isfaaghyth thank youu

Reply:
Hi @isfaaghyth I've created single factory for the ViewHolders with your repo as reference, I tweak it a little bit on this commit 0f2fed9

Please kindly to review it again 🙏 🙏

Reply:
thanks, @pandubaraja. awesome!